rotkehlchen/tests/conftest.py

```diff
@@ -138,3 +140,44 @@ def requires_env(allowed_envs: list[TestEnvironment]):
         'CI' in os.environ and env not in allowed_envs,
         reason=f'Not suitable envrionment {env} for current test',
     )
+
+
+def get_cassette_dir(request: pytest.FixtureRequest) -> str:
+    return request.node.module.__name__
+
+
+@pytest.fixture
+def vcr_config(request: pytest.FixtureRequest, default_cassette_name: str) -> dict[str, Any]:
+    """
+    vcrpy config
+    - record mode: allow rewriting multiple cassettes using CASSETTE_REWRITE_PATH env variable
+      e.g. CASSETTE_REWRITE_PATH=rotkehlchen.tests.unit.decoders will rewrite all decoder cassettes
+    - decode compressed response: decode if not in CI (for readability)
+    """
+    rewrite_path = os.environ.get('CASSETTE_REWRITE_PATH')
+    cassette_dir = get_cassette_dir(request)
+    cassette_path = os.path.join(cassette_dir, default_cassette_name)
+
+    record_mode = "rewrite" if rewrite_path and cassette_path.startswith(rewrite_path) else "once"
+    decode_compressed_response = 'CI' not in os.environ
+
+    return {
+        "record_mode": record_mode,
+        "decode_compressed_response": decode_compressed_response,
+    }
+
+
+@pytest.fixture(scope="module")
+def vcr_cassette_dir(request: pytest.FixtureRequest) -> str:
```

Comment:
By default `pytest-recording` [stores](https://github.com/kiwicom/pytest-recording/blob/v0.12.1/src/pytest_recording/plugin.py#L159) cassettes alongside test source code, which is probably not ideal for us

Reply:
We can further update this directory structure to nest test cases inside a test directory inside the module path, if needed.

Or.. should we create a fully nested directory structure?  - `rotkehlchen/tests/unit/decoders`

Open to discussion.

Reply:
About the directory structure I would say that there is no need to have `rotkehlchen/` directory and `tests/` directory since all tests are located under `rotkehlchen/tests`, but for other directories (`unit/`, `decoders/`, etc.) I think that makes sense to have them nested. Otherwise we will just have a ton of files in a single dir, which doesn't sound nice.

Reply:
updated to this

```
cassettes
└── unit
    └── decoders
        ├── test_aave
        │   ├── test_aave_deposit_v1[ethereum_accounts0].yaml
        │   ├── test_aave_eth_withdraw_v1[ethereum_accounts0].yaml
        │   └── test_aave_withdraw_v1[ethereum_accounts0].yaml
        ├── test_compound
        │   ├── test_compound_deposit_with_comp_claim[ethereum_accounts0].yaml
        │   ├── test_compound_ether_deposit[ethereum_accounts0].yaml
        │   └── test_compound_ether_withdraw[ethereum_accounts0].yaml
        ...
```

Reply:
Yeah, sounds good :+1: